sklearnex/covariance/incremental_covariance.py

```diff
@@ -0,0 +1,132 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import numpy as np
+from sklearn.utils import check_array, gen_batches
+
+from daal4py.sklearn._utils import control_n_jobs, run_with_n_jobs
+from onedal._device_offload import support_usm_ndarray
+from onedal.covariance import (
+    IncrementalEmpiricalCovariance as onedal_IncrementalEmpiricalCovariance,
+)
+
+
+@control_n_jobs
+class IncrementalEmpiricalCovariance:
+    """
+    Incremental estimator for covariance.
+    Allows to compute empirical covariance estimated by maximum
+    likelihood method if data are splitted into batches.
+
+    Parameters
+    ----------
+    batch_size : int, default=None
+        The number of samples to use for each batch. Only used when calling
+        ``fit``. If ``batch_size`` is ``None``, then ``batch_size``
+        is inferred from the data and set to ``5 * n_features``, to provide a
+        balance between approximation accuracy and memory consumption.
+
+    Attributes
+    ----------
+    location_ : ndarray of shape (n_features,)
+        Estimated location, i.e. the estimated mean.
+
+    covariance_ : ndarray of shape (n_features, n_features)
+        Estimated covariance matrix
+    """
+
+    _onedal_incremental_covariance = staticmethod(onedal_IncrementalEmpiricalCovariance)
+
+    def __init__(self, batch_size=None):
+        self._need_to_finalize = False  # If True then finalize compute should
+        #      be called to obtain covariance_ or location_ from partial compute data
+        self.batch_size = batch_size
+
+    @run_with_n_jobs
+    def _onedal_finalize_fit(self):
+        assert hasattr(self, "_onedal_estimator")
+        self._onedal_estimator.finalize_fit()
+        self._need_to_finalize = False
+
+    @run_with_n_jobs
+    def _onedal_partial_fit(self, X, queue):
+        onedal_params = {
+            "method": "dense",
+            "bias": True,
+        }
+        if not hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator = self._onedal_incremental_covariance(**onedal_params)
+        self._onedal_estimator.partial_fit(X, queue)
+        self._need_to_finalize = True
+
+    @property
+    def covariance_(self):
+        if self._need_to_finalize:
+            self._onedal_finalize_fit()
+        return self._onedal_estimator.covariance_
+
+    @property
+    def location_(self):
+        if self._need_to_finalize:
+            self._onedal_finalize_fit()
+        return self._onedal_estimator.location_
+
+    @support_usm_ndarray()
+    def partial_fit(self, X, queue=None):
```

Comment:
This is not a scope of the current PR, it is required some follow up work here.
queue param here is only used for `support_usm_ndarray`.
It is better update to `support_usm_ndarray` and update passing queue into onedal4py backend.